### DISCUSSION TITLE
请问ecapture是如何设计并存储长度较大的变量值的？

### DISCUSSION THREAD

**Discussion by Your7Maxx** [gojue/ecapture#337] [Q&A] [answered]
在提问之前，我阅读了`ecapture`项目的源码，遗憾的是我并没有清晰地找到`ecapture`是如何设计并存储长度较大变量值的方式。这个问题来源于我在进行开发测试时，总是会遇到来自`verifier`的报错：`Looks like the BPF stack limit of 512 bytes is exceeded. Please move large on stack variables into BPF per-cpu array map.`，而提示信息给我的解决方法`BPF_PERCPU_ARRAY`并不适配我的程序。举个例子，在从SSL缓冲区读取请求信息时，我的做法总是先定义一个`data_t`结构体，在`data_t`结构体中定义`char request[MAX_BUF_SIZE]`成员字段用来存储从SSL缓冲区获取到的请求信息，`verifier`是否会报错取决于程序对`MAX_BUF_SIZE`的大小定义，这显然是比较拙略的设计方式，并且面对长度较大的请求便会出现截断现象。在测试之后，我发现`ecaputre`并没有出现过类似的问题，对于长度较大的请求也能很好的处理，所以想来和作者请教一下~

**Comment by cfc4n** [comment, marked as answer]
可以看下kern目录下的make_event函数

**Reply by Your7Maxx** [reply]
Got it！在这个话题下问一个不是特别相关的问题：`ecapture`什么时候可以支持对`go`语言程序HTTPS请求的捕获呀？或者说这方面的实现思路是怎么样的呢？期待您的分享！

**Reply by cfc4n** [reply]
已经有部份能力了。可以看下项目中gotls相关文件。
